gtest/src/lib.rs

```diff
@@ -28,227 +28,311 @@
+//! gstd = { git = "https://github.com/gear-tech/gear.git", tag = "v1.0.1" }
 //!
 //! [build-dependencies]
-//! gear-wasm-builder = { git = "https://github.com/gear-tech/gear.git", tag = "v1.0.0" }
+//! gear-wasm-builder = { git = "https://github.com/gear-tech/gear.git", tag = "v1.0.1" }
 //!
 //! [dev-dependencies]
-//! gtest = { git = "https://github.com/gear-tech/gear.git", tag = "v1.0.0" }
+//! gtest = { git = "https://github.com/gear-tech/gear.git", tag = "v1.0.1" }
 //! ```
 //!
-//! ## `gtest` capabilities
+//! ## Program example
 //!
-//! - Initialization of the common environment for running smart contracts:
-//! ```ignore
-//! // This emulates node's and chain's behavior.
-//! //
-//! // By default, sets:
-//! // - current block equals 0
-//! // - current timestamp equals UNIX timestamp of your system.
-//! // - minimal message id equal 0x010000..
-//! // - minimal program id equal 0x010000..
+//! Let's write a simple program that will receive a message and reply to it.
+//!
+//! `lib.rs`:
+//!
+//! ```
+//! #![no_std]
+//! use gstd::msg;
+//!
+//! #[no_mangle]
+//! extern "C" fn handle() {
+//!     let payload = msg::load_bytes().expect("Failed to load payload");
+//!
+//!     if payload == b"PING" {
+//!         msg::reply_bytes(b"PONG", 0).expect("Failed to send reply");
+//!     }
+//! }
+//! ```
+//!
+//! `build.rs`:
+//!
+//! ```
+//! fn main() {
+//!     gear_wasm_builder::build();
+//! }
+//! ```
+//!
+//! We will add a test that will check the program's behavior. To do this, we
+//! will use the `gtest` library.
+//!
+//! Our test will consist of the following steps:
+//!
+//! 1. Initialize the `System` structure.
+//! 2. Initialize the `Program` structure.
+//! 3. Send an init message to the program. We don't have an `init` function in
+//!    our program, but the first message to the program is always the init
+//!    message while using `gtest`.
+//! 4. Send a handle message to the program.
+//! 5. Check the result of the program execution.
+//!
+//! Add these lines to the bottom of the `lib.rs` file:
+//!
+//! ```
+//! #[cfg(test)]
+//! mod tests {
+//!     use gtest::{Log, Program, System};
+//!
+//!     const USER_ID: u64 = 100001;
+//!
+//!     #[test]
+//!     fn test_ping_pong() {
+//!         // Initialization of the common environment for running smart contracts.
+//!         let sys = System::new();
+//!
+//!         // Initialization of the current program structure.
+//!         let prog = Program::current(&sys);
+//!
+//!         // Send an init message to the program.
+//!         let res = prog.send_bytes(USER_ID, b"Doesn't matter");
+//!
+//!         // Check whether the program was initialized successfully.
+//!         assert!(!res.main_failed());
+//!
+//!         // Send a handle message to the program.
+//!         let res = prog.send_bytes(USER_ID, b"PING");
+//!
+//!         // Check the result of the program execution.
+//!         // 1. Create a log with the expected result.
```

Comment:
```suggestion
//!         // 1. Create a log pattern with the expected result.
```
??? As this doesn't seem to be a log

Reply:
Thank you 🙏 
Fixed in 233adecc